src-control/Control/RefCount.hs

```diff
@@ -219,7 +219,7 @@ class RefCounted obj where
 --
 newRef ::
      (RefCounted obj, FinaliserM obj ~ m, PrimMonad m)
-     HAS_CALL_STACK
+  => HasCallStackIfDebug
```

Comment:
What's the reason that this does not throw a warning?

Reply:
By the looks of it, GHC combines successive constraint arrows, so by the time it reaches the warning subsystem, `(A, B) => () => x` has turned into `(A, B) => x` by concatenation, whereas it does no other normalisation, so `(A, B, ()) => x` remains unchanged.